pkg/generate/app/cmd/resolve.go

```diff
@@ -108,7 +106,10 @@ func Resolve(r *Resolvers, c *ComponentInputs, g *GenerationInputs) (*ResolvedCo
 	for _, repo := range repositories {
 		repo.SetContextDir(g.ContextDir)
 	}
-
+	// Environment File where the parameters are stored
+	if len(g.EnvironmentFile) != 0 && !app.IsPossibleEnvironmentFile(g.EnvironmentFile) {
+		return nil, fmt.Errorf("when --env-file is specified you must provide a valid env file location")
+	}
```

Comment:
i think you are misunderstanding what this function (Resolve) is for and when it is called.  It is to determine, when you run "oc new-app A B C D" what A, B, C, and D are (eg are they template names?  image names? imagestream names?  names of template files that exist on disk?  git repository names?).  A, B, C, and D will never by an "environment file" because environment files will be specified explicitly as "--env-file myenvfile"

So Resolve is not to handle this new "environmentfile" which is an explicit argument that we can validate and process as part of the CLI setup/validation.

Validating that the environment file exists and contains valid content should be done in either validate() or AddArguments(), within pkg/generate/app/cmd/newapp.go